Orange/canvas/application/errorreporting.py

```diff
@@ -38,6 +38,20 @@ class OWWidget: pass
 log = logging.getLogger()
 
 
+def get_developer_id():
+    """Check if developer provided his name for Sentry reports."""
+    try:
+        from Orange.misc.environ import data_dir_base
+        fname = os.path.join(data_dir_base(), 'Orange', 'developer_id.txt')
```

Comment:
The case for environment variables vs. just program arguments is that the behavior is tied to the environment instead of program invocation. A developer, regardless of if I run Orange from terminal, through PyCharm, through gdb or valgrind, or if I run standalone widgets sans canvas, I want Orange to expose the debug interface, whatever that be, and to object loudly to deprecated interfaces. But I can _add_ program arguments for those, if you wish.

> causes tests to fail locally if it is not set

That's a bug. :+1: